orderer/common/cluster/deliver.go

```diff
@@ -614,3 +614,54 @@ type errorAndResponse struct {
 	err  error
 	resp *orderer.DeliverResponse
 }
+
+// TODO: This message would be removed after completion
+// BlockFetcher is a wrapper over BlockPuller which can pull blocks from an orderer endpoint
+// besides being able to terminate connection to a malicious orderer and connect to another ordering node
+// for bft capability.
+type BlockFetcher struct {
+	BlockPuller
+	BFTTimeOutChannel chan struct{} // Not used now, a placeholder
+}
+
+func (p *BlockFetcher) probeAndConnect(endpoint EndpointCriteria, seq uint64) {
+
+	ei, err := p.probeEndpoint(endpoint, seq)
+	if err != nil {
+		p.Logger.Warningf("Received error of type '%v' from %s", err, endpoint.Endpoint)
+		p.Logger.Debugf("%s's TLSRootCAs are %s", endpoint.Endpoint, endpoint.TLSRootCAs)
+		if err == ErrForbidden {
+			//atomic.StoreUint32(&forbiddenErr, 1)
+		}
+		if err == ErrServiceUnavailable {
+			//atomic.StoreUint32(&unavailableErr, 1)
+		}
+
+	}
+
+	// Connect to this endpoint
+	p.conn = ei.conn
+	p.endpoint = ei.endpoint
+	p.latestSeq = ei.lastBlockSeq
+}
+
+func (p *BlockFetcher) PullBlock(seq uint64) *common.Block {
+	// mechanism to know the type of consensus: bft or cft
+	// if cft, just pull blocks
+	// return p.BlockPuller.PullBlock
+
+	// in case of bft, initialise a connection to single endpoint
+	for _, endpoint := range p.Endpoints {
+		p.probeAndConnect(endpoint, seq)
+		if !p.isDisconnected() {
+			block := p.BlockPuller.tryFetchBlock(seq)
```

Comment:
why are we calling concrete methods of block puller? it should be defined as an interface that we consume.